### PR TITLE
[PM-9953][PM-9963] Collections Reprompt and Org assignment

### PR DESCRIPTION
**Pull request by nick-livefront**
## 🎟️ Tracking

[PM-9953](https://bitwarden.atlassian.net/browse/PM-9953)
[PM-9963](https://bitwarden.atlassian.net/browse/PM-9963)

## 📔 Objective

PM-9953 - reprompt the user for their password password before navigating to the collection assignment component (when applicable)
PM-9963 - Only pass collections that are assigned to the cipher's org (when applicable)

## 📸 Screenshots

|Password Reprompt|
|-|
|<video src="https://github.com/user-attachments/assets/0b6f345e-0562-4bee-8c06-e8cd1094bca4" />|

|Collections - Before| Collections - After|
|-|-|
|<video src="https://github.com/user-attachments/assets/157f11f7-f9fb-4300-b9a0-20ca3a58cd7c"/>|<video src="https://github.com/user-attachments/assets/4b1faa61-35ee-42fd-bda7-7287092932a9"/>|

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-9953]: https://bitwarden.atlassian.net/browse/PM-9953?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PM-9963]: https://bitwarden.atlassian.net/browse/PM-9963?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Jingo88 — APPROVED** [review]

File: apps/browser/src/vault/popup/components/vault-v2/item-more-options/item-more-options.component.html [hunk 1]
```diff
@@ -28,12 +28,7 @@
       <a routerLink="" bitMenuItem (click)="clone()">
         {{ "clone" | i18n }}
       </a>
-      <a
-        routerLink="/assign-collections"
-        [queryParams]="{ cipherId: this.cipher.id }"
-        type="button"
-        bitMenuItem
-      >
+      <a type="button" bitMenuItem (click)="conditionallyNavigateToAssignCollections()">
```
Comment (shane-melton) [inline apps/browser/src/vault/popup/components/vault-v2/item-more-options/item-more-options.component.html:31]: ⛏️ Anchor tags don't support the `type="button"` attribute. (I know it was already there, but we clean it up while we're here)

**Review by shane-melton — CHANGES_REQUESTED** [review]
Looks good, just one tiny nit.

File: apps/browser/src/vault/popup/components/vault-v2/item-more-options/item-more-options.component.html (hunk 1 above)
Comment (nick-livefront) [reply apps/browser/src/vault/popup/components/vault-v2/item-more-options/item-more-options.component.html]: Good catch! [9080539](https://github.com/bitwarden/clients/pull/10194/commits/908053964c9c87227f7604382fc7f5095896dfdc)

**Comment by nick-livefront** [comment]
Apologies, @dani-garcia I accidentally pushed a wrong file. I have since removed, no need for your review.

**Review by Jingo88 — APPROVED** [review]

**Comment by nick-livefront** [comment]
@shane-melton When you get a chance could you take a look at this one? Since you requested changes, I need your approval

**Review by shane-melton — APPROVED** [review]

**Pull request merged**